lib/msal-node/test/network/HttpClient.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { HttpClient } from '../../src/network/HttpClient';
+import axios, { AxiosResponse } from 'axios';
+import { mocked } from "ts-jest/dist/util/testing"; //<-- This allows to mock results
+
+jest.mock('axios');
+
+describe('HttpClient', () => {
+
+    // instantiate httpClient
+    const httpClient = new HttpClient();
+
+    // Mock successful response
+    const axiosResponse: AxiosResponse = {
+        data: {
+            title: 'mock axios response',
+            body: 'Well .. this is mock data',
+        },
+        status: 200,
+        statusText: 'OK',
+        config: {},
+        headers: {},
+    };
+
+    // Mock error response
+    const axiosErrorResponse = {
+        message: 'axios request failed',
+        response: {
+            data: {
+                errorCode: 'invalid_client',
+                errorSummary: "Invalid value for 'client_id' parameter.",
+            },
+            status: 400,
+            statusText: 'Bad Request',
+            headers: {},
+            config: {},
+        },
+    };
+
+    // test GET
+    test('sendGetRequestAsync', async () => {
+
+        //Mocking axios function rather than a method
+        mocked(axios).mockResolvedValue(axiosResponse);
```

Comment:
This should be done in `before` inside each test.

Reply:
beforeEach() and mock do not seem to work together, it fails. I did some digging and have seen only examples resetting the mock in before it, not add the mock. I will get back to this when I have more time. Keeping it as is for now.

Reply:
Ah interesting, good to know.